### PR TITLE
Integration Test Target Setup

### PR DESCRIPTION
**Pull request by aashishpatil-g**
Adds an Integration Test Target. 

Tests cannot be run within a git action as yet but I've run them manually locally and they are executing successfully.

### CODE REVIEW

**Comment by paulb777** [comment]
Looks like copyrights are getting deleted

**Comment by aashishpatil-g** [comment]
> Looks like copyrights are getting deleted

Sorry about that. Missed them in the generated code. Added now.

**Comment by paulb777** [comment]
Consider using an environment variable like in #8 to wrap integration tests so CI can succeed.  

I'll investigate working around the CI issue in another PR with less parallelization

**Comment by aashishpatil-g** [comment]
> Consider using an environment variable like in #8 to wrap integration tests so CI can succeed.
> 
> I'll investigate working around the CI issue in another PR with less parallelization

Added a flag to disable integration tests and added this flag to the spm.yml

File: Tests/Integration/Gen/KitchenSink/Sources/KitchenSinkClient.swift [hunk 1]
```diff
@@ -15,15 +15,13 @@
 import FirebaseDataConnect
 import Foundation
 
-@available(macOS 10.15, iOS 13, tvOS 13, watchOS 6, *)
```
Comment (paulb777) [inline Tests/Integration/Gen/KitchenSink/Sources/KitchenSinkClient.swift:18]: Was it intentional to remove these? Shouldn't the Package.swift versions be updated to these in conjunction with this removal?

**Review by paulb777 — APPROVED** [review]

File: Tests/Integration/Gen/KitchenSink/Sources/KitchenSinkClient.swift (hunk 1 above)
Comment (aashishpatil-g) [reply Tests/Integration/Gen/KitchenSink/Sources/KitchenSinkClient.swift:18]: The style script seems to have removed these. They are part of the generated code.

Comment (paulb777) [reply Tests/Integration/Gen/KitchenSink/Sources/KitchenSinkClient.swift:18]: Hmm, the style script isn't touching those when I run it ...

Comment (aashishpatil-g) [reply Tests/Integration/Gen/KitchenSink/Sources/KitchenSinkClient.swift:18]: Sorry, I think they got stripped out when I regenerated the code. The code is not part of the core SDK. Its generated so we can run the integration tests.

Comment (aashishpatil-g) [reply Tests/Integration/Gen/KitchenSink/Sources/KitchenSinkClient.swift:18]: We have the version gates on the tests and core sdk so the gate on the gen code isn't required.

**Pull request merged**